contracts/MarginCalculator.sol

```diff
@@ -2,11 +2,280 @@
+        int256 _shortAmount,
+        int256 _longAmount,
+        int256 _shortStrike,
+        int256 _longStrike
+    ) internal pure returns (int256) {
+        return _longStrike.mul(FixedPointInt256.min(_shortAmount, _longAmount)).sub(_shortAmount.mul(_shortStrike));
+    }
+
+    /**
+     * @dev calculate non-expired net call spread value.
+     *                                                min (long amount, short amount) * max (0, long strike - short strike)
+     * net = min(0, long amount - short amount) -  --------------------------------------------------------------------------
+     *                                                                             long strike
+     *
+     * @dev if long strike = 0 (no long token), then return net = short amount.
+     * @return net value
+     */
+    function _calculateNonExpiredCallNetValue(
+        int256 _shortAmount,
+        int256 _longAmount,
+        int256 _shortStrike,
+        int256 _longStrike
+    ) internal pure returns (int256) {
+        if (_longStrike == 0) {
+            return -_shortAmount;
+        }
+        return
+            FixedPointInt256.min(0, _longAmount.sub(_shortAmount)).sub(
+                FixedPointInt256
+                    .min(_longAmount, _shortAmount)
+                    .mul(FixedPointInt256.max(0, _longStrike.sub(_shortStrike)))
+                    .div(_longStrike)
+            );
+    }
+
+    /**
+     * @dev calculate expired net put spread value.
+     *
+     * Formula: net = ( long cash value * long Amount ) - (short cash value * short amount)
+     *
+     * @return net value
+     */
+    function _calculateExpiredPutNetValue(
+        int256 _shortAmount,
+        int256 _longAmount,
+        int256 _shortCashValue,
+        int256 _longCashValue
+    ) internal pure returns (int256) {
+        return (_longCashValue.mul(_longAmount)).sub(_shortCashValue.mul(_shortAmount));
+    }
+
+    /**
+     * @dev calculate expired net call spread value.
+     *                     ( long cash value * long Amount ) - (short cash value * short amount)
+     *  Formula: net =   -------------------------------------------------------------------------
+     *                                               Underlying price
+     * @return net value
+     */
+    function _calculateExpiredCallNetValue(
+        int256 _shortAmount,
+        int256 _longAmount,
+        int256 _shortCashValue,
+        int256 _longCashValue,
+        int256 _underlyingPriceInt
+    ) internal pure returns (int256) {
+        return (_longCashValue.mul(_longAmount)).sub(_shortCashValue.mul(_shortAmount)).div(_underlyingPriceInt);
+    }
+
+    /**
+     * @dev internal function that ensure each asset type & amout array have the same length, and length <= 1;
+     * @param _vault the vault to check.
+     */
+    function _checkAssetCount(Vault memory _vault) internal pure {
+        // For the currect version, check lengths of short, long, ollateral <= 1.
```

Comment:
"collateral" spelling

Reply:
"current"

Reply:
is there a way to check spelling everywhere?

Reply:
I think so, let me find that plugin 🤣